modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/FileEntryVideoEmbeddableHTMLItemSelectorReturnTypeResolver.java

```diff
@@ -58,6 +61,23 @@ public String getValue(FileEntry fileEntry, ThemeDisplay themeDisplay)
 			}
 		).put(
 			"title", fileEntry.getTitle()
+		).put(
+			"url",
+			() -> {
+				if (!FeatureFlagManagerUtil.isEnabled("LPD-11235")) {
+					return null;
+				}
+
+				DLVideoExternalShortcut dlVideoExternalShortcut =
+					_dlVideoExternalShortcutResolver.resolve(
+						fileEntry.getFileVersion());
+
+				if (dlVideoExternalShortcut != null) {
+					return dlVideoExternalShortcut.getURL();
+				}
+
+				return null;
+			}
```

Comment:
On selection, we are exposing video URL, along with final HTML that is used by CKE4. The complex logic of different HTML for different hosting providers, along with support for preview, is entirely left to [CKE5 Media embed plugin](https://ckeditor.com/docs/ckeditor5/latest/features/media-embed.html).

This code works in manual testing (see video in opening comment), but for some reason fails in Playwright test (`test.step('Select a video from document library')`). It returns an empty string for `html` and `url`. I cannot explain it, will need help, maybe from content management team.